src/capi/typeobject.cpp

```diff
@@ -1729,7 +1729,9 @@ extern "C" int PyType_Ready(PyTypeObject* cls) noexcept {
 
     RELEASE_ASSERT(cls->tp_getattro == NULL || cls->tp_getattro == PyObject_GenericGetAttr, "");
     RELEASE_ASSERT(cls->tp_setattro == NULL || cls->tp_setattro == PyObject_GenericSetAttr, "");
-    RELEASE_ASSERT(cls->tp_as_buffer == NULL, "");
+
+    // TODO: I think this is not implemented yet but the arraymodules uses it.
+    // RELEASE_ASSERT(cls->tp_as_buffer == NULL, "");
```

Comment:
Sorry, let me rephrase -- most of the tp_* methods are for a class to request non-default behavior for itself, so for something like tp_getattro, if we ignored that being set we'd start silently doing the wrong thing for that class.  For tp_as_buffer, I think it's ok to ignore it, because it doesn't really have a 'default behavior' that it overrides.  So verifying isn't so much checking that a specific class's tp_as_buffer works, but that classes get the behavior that they expect when they set it.  I think in this particular case, it's ok because we haven't implemented anything that deals with tp_as_buffer, so we can just remove the check.